### PR TITLE
allowDynamicKeys

### PR DESCRIPTION
**Pull request by delphiactual**
#### Checklist

- [X] only relevant code is changed (make a diff before you submit the PR)
- [X] run tests `npm run test`
- [X] tests are included

image

after updated tests
image

### CODE REVIEW

**Comment by delphiactual** [comment]
By utilizing allowDynamicKeys and metadata, dynamic keys can be inferred using a custom handler.

File: test/parser.js [hunk 1]
```diff
@@ -1090,14 +1090,15 @@ test('Extract properties from template literals', (t) => {
     const parser = new Parser({
         defaultValue: function(lng, ns, key) {
             if (lng === 'en') {
-                return key;
+                return key.replace(/\r\n/g, '\n');
```
Comment (delphiactual) [inline test/parser.js:1093]: this makes this test OS agnostic

File: test/parser.js [hunk 2]
```diff
@@ -1090,14 +1090,15 @@ test('Extract properties from template literals', (t) => {
...
             }
             return '__NOT_TRANSLATED__';
         },
         keySeparator: false,
-        nsSeparator: false
+        nsSeparator: false,
+        allowDynamicKeys: false
     });
-    const content = fs.readFileSync(path.resolve(__dirname, 'fixtures/template-literals.js'), 'utf8');
+    const content = fs.readFileSync(path.resolve(__dirname, 'fixtures/template-literals.js'), 'utf8').replace(/\r\n/g, '\n');
```
Comment (delphiactual) [inline test/parser.js:1101]: same here

**Comment by cheton** [comment]
It looks good to me.

Do you have a more complete test case and test fixtures that can help people understand how `allowDynamicKeys` works? Thank you.

**Comment by delphiactual** [comment]
@cheton I have added a more complete test case and included a sample transform function showing usage.

**Comment by cheton** [comment]
v3.3.0 was published. Thank you for making the improvement.

**Pull request merged**